src/fs/auth.go

```diff
@@ -15,31 +15,40 @@ func use(h http.HandlerFunc, middleware ...func(http.HandlerFunc) http.HandlerFu
 	return h
 }
 
+func isAdmin(r *http.Request) bool {
+	// if Authorization header is not present, this is admin user
+	authToken := r.Header.Get("Authorization")
+	return authToken == ""
+}
+
 func (service *MercuryFsService) authenticate(writer http.ResponseWriter, request *http.Request) {
+	// decode and parse json request body
 	decoder := json.NewDecoder(request.Body)
 	data := make(map[string]interface{})
 	err := decoder.Decode(&data)
 	if err != nil {
 		panic(err)
```

Comment:
does this mean that client data in json (malformed) will crash our service?

Reply:
Thanks for pointing it out. I have corrected it.